modules/unsupported/wfs-ng/src/test/java/org/geotools/data/wfs/online/KartverketStedsnavnDataStoreOnlineTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.wfs.online;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import javax.xml.namespace.QName;
+import org.geotools.data.wfs.WFSDataStoreFactory;
+import org.geotools.data.wfs.WFSTestData.TestDataType;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.locationtech.jts.geom.Polygon;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.GeometryDescriptor;
+import org.opengis.filter.Filter;
+import org.opengis.filter.Id;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/** @author Roar Brænden */
+public class KartverketStedsnavnDataStoreOnlineTest extends AbstractWfsDataStoreOnlineTest {
+
+    static final String SERVER_URL =
+            "https://wfs.geonorge.no/skwms1/wfs.stedsnavn?request=GetCapabilities&service=WFS";
```

Comment:
Thanks for an online test case to verify the functionality.

It may be worth covering this in a static get capabilities test case if you run into any issues in the future (as online tests are not often run).

Reply:
Either that or we could have an online test actually being run as a PR check, similar to what is done for postgresql and the like. Only worry is reliability: in the case of databases we set them up and control them, here the services are managed by external entities, tests might start failing either randomly or even permanently due to changes to their configs.

Overall having a mock test that relies on saved documents seems better, it can be always run and it's reliable.

Reply:
Well, since there is plenty of online tests already anyways, we can leave "moving to a mock test" to a future PR. Merging.